frame/elections-phragmen/src/lib.rs

```diff
@@ -195,29 +326,40 @@ pub trait Trait: frame_system::Trait {
 decl_storage! {
 	trait Store for Module<T: Trait> as PhragmenElection {
 		// ---- State
-		/// The current elected membership. Sorted based on account id.
-		pub Members get(fn members): Vec<(T::AccountId, BalanceOf<T>)>;
-		/// The current runners_up. Sorted based on low to high merit (worse to best).
-		pub RunnersUp get(fn runners_up): Vec<(T::AccountId, BalanceOf<T>)>;
+		/// The current elected members.
+		///
+		/// Invariant: Always sorted based on account id.
+		pub Members get(fn members): Vec<SeatHolder<T::AccountId, BalanceOf<T>>>;
+
+		/// The current reserved runners-up.
+		///
+		/// Invariant: Always sorted based on rank (worse to best). Upon removal of a member, the
+		/// last (i.e. _best_) runner-up will be replaced.
+		pub RunnersUp get(fn runners_up): Vec<SeatHolder<T::AccountId, BalanceOf<T>>>;
+
 		/// The total number of vote rounds that have happened, excluding the upcoming one.
 		pub ElectionRounds get(fn election_rounds): u32 = Zero::zero();
 
 		/// Votes and locked stake of a particular voter.
 		///
-		/// TWOX-NOTE: SAFE as `AccountId` is a crypto hash
-		pub Voting get(fn voting): map hasher(twox_64_concat) T::AccountId => (BalanceOf<T>, Vec<T::AccountId>);
+		/// TWOX-NOTE: SAFE as `AccountId` is a crypto hash.
+		pub Voting get(fn voting): map hasher(twox_64_concat) T::AccountId => Voter<T::AccountId, BalanceOf<T>>;
 
-		/// The present candidate list. Sorted based on account-id. A current member or runner-up
+		/// The present candidate list. A current member or runner-up
 		/// can never enter this vector and is always implicitly assumed to be a candidate.
-		pub Candidates get(fn candidates): Vec<T::AccountId>;
+		///
+		/// Second element is the deposit.
+		///
+		/// Invariant: Always sorted based on account id.
+		pub Candidates get(fn candidates): Vec<(T::AccountId, BalanceOf<T>)>;
 	} add_extra_genesis {
 		config(members): Vec<(T::AccountId, BalanceOf<T>)>;
 		build(|config: &GenesisConfig<T>| {
 			let members = config.members.iter().map(|(ref member, ref stake)| {
-				// make sure they have enough stake
```

Comment:
Given that you now correctly track user's deposit, I don't see why this is a requirement. Just have the initial members have zero stake.

I think it may even be weird for people that genesis will try to reserve some value.

Reply:
kk -- genesis members will vote for themselves, with _**no deposit**_ and _**no locks**_. Kind of a fishy situation but I see also your point.

Reply:
Why initial members wouldn't need to have a correct deposit and locks ? I prefer to panic on genesis and having a more sane state than the opposite.

Reply:
It is the same thing as what we do with a lot of genesis state, like indices for example. All genesis indices have zero deposit. Its strange to me that genesis will be doing things to manipulate locks and balances, but maybe it is fine.

Reply:
I don't have a strong opinion on this so I will keep it as it is. 

My personal preference is what @thiolliere recommends, being accurate. 
But I think overall it is better for substrate pallets to be consistent and behave similarly. Given the indices example, not reserving makes more sense.

Reply:
ok then I'm fine with any ways.